tests/optional_scalars_generated.js

```diff
@@ -0,0 +1,782 @@
+// automatically generated by the FlatBuffers compiler, do not modify
+
+/**
+ * @const
+ * @namespace
+ */
+var optional_scalars = optional_scalars || {};
+
+/**
+ * @enum {number}
+ */
+optional_scalars.OptionalByte = {
+  None: 0,
+  One: 1,
+  Two: 2
+};
+
+/**
+ * @enum {string}
+ */
+optional_scalars.OptionalByteName = {
+  '0': 'None',
+  '1': 'One',
+  '2': 'Two'
+};
+
+/**
+ * @constructor
+ */
+optional_scalars.ScalarStuff = function() {
+  /**
+   * @type {flatbuffers.ByteBuffer}
+   */
+  this.bb = null;
+
+  /**
+   * @type {number}
+   */
+  this.bb_pos = 0;
+};
+
+/**
+ * @param {number} i
+ * @param {flatbuffers.ByteBuffer} bb
+ * @returns {optional_scalars.ScalarStuff}
+ */
+optional_scalars.ScalarStuff.prototype.__init = function(i, bb) {
+  this.bb_pos = i;
+  this.bb = bb;
+  return this;
+};
+
+/**
+ * @param {flatbuffers.ByteBuffer} bb
+ * @param {optional_scalars.ScalarStuff=} obj
+ * @returns {optional_scalars.ScalarStuff}
+ */
+optional_scalars.ScalarStuff.getRootAsScalarStuff = function(bb, obj) {
+  return (obj || new optional_scalars.ScalarStuff).__init(bb.readInt32(bb.position()) + bb.position(), bb);
+};
+
+/**
+ * @param {flatbuffers.ByteBuffer} bb
+ * @param {optional_scalars.ScalarStuff=} obj
+ * @returns {optional_scalars.ScalarStuff}
+ */
+optional_scalars.ScalarStuff.getSizePrefixedRootAsScalarStuff = function(bb, obj) {
+  bb.setPosition(bb.position() + flatbuffers.SIZE_PREFIX_LENGTH);
+  return (obj || new optional_scalars.ScalarStuff).__init(bb.readInt32(bb.position()) + bb.position(), bb);
+};
+
+/**
+ * @param {flatbuffers.ByteBuffer} bb
+ * @returns {boolean}
+ */
+optional_scalars.ScalarStuff.bufferHasIdentifier = function(bb) {
+  return bb.__has_identifier('NULL');
+};
+
+/**
+ * @returns {number}
+ */
+optional_scalars.ScalarStuff.prototype.justI8 = function() {
+  var offset = this.bb.__offset(this.bb_pos, 4);
+  return offset ? this.bb.readInt8(this.bb_pos + offset) : 0;
+};
+
+/**
+ * @returns {number|null}
+ */
+optional_scalars.ScalarStuff.prototype.maybeI8 = function() {
+  var offset = this.bb.__offset(this.bb_pos, 6);
+  return offset ? this.bb.readInt8(this.bb_pos + offset) : null;
```

Comment:
hurray for dynamically typed languages, they make a `null` integer rather simple :)